compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -448,11 +448,16 @@ impl<'a> Parser<'a> {
             })
         }
 
-        let mut expected = edible
+        self.expected_tokens.extend(
+            edible
+                .iter()
+                .map(|x| TokenType::Token(x.clone()))
+                .chain(inedible.iter().map(|x| TokenType::Token(x.clone()))),
+        );
```

Comment:
This is necessary because otherwise `token::Gt` won't show up in `expected_tokens` when we check for it when emitting the diagnostic.